tests/components/kostal_plenticore/test_number.py

```diff
@@ -1,70 +1,105 @@
 """Test Kostal Plenticore number."""
 
-from unittest.mock import AsyncMock, MagicMock
+from collections.abc import Generator
+from datetime import timedelta
+from unittest.mock import AsyncMock, MagicMock, patch
 
-from kostal.plenticore import SettingsData
+from kostal.plenticore import PlenticoreApiClient, SettingsData
 import pytest
 
-from homeassistant.components.kostal_plenticore.number import (
-    PlenticoreNumberEntityDescription,
+from homeassistant.components.number import (
+    ATTR_VALUE,
+    DOMAIN as NUMBER_DOMAIN,
+    SERVICE_SET_VALUE,
 )
+from homeassistant.components.number.const import ATTR_MAX, ATTR_MIN
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_registry import async_get
+from homeassistant.util import dt
 
-from tests.common import MockConfigEntry
+from tests.common import MockConfigEntry, async_fire_time_changed
 
 
 @pytest.fixture
-def mock_coordinator() -> MagicMock:
-    """Return a mocked coordinator for tests."""
-    coordinator = MagicMock()
-    coordinator.async_write_data = AsyncMock()
-    coordinator.async_refresh = AsyncMock()
-    return coordinator
+def mock_plenticore_client() -> Generator[PlenticoreApiClient, None, None]:
+    """Return a patched PlenticoreApiClient."""
+    with patch(
+        "homeassistant.components.kostal_plenticore.helper.PlenticoreApiClient",
+    ) as plenticore_client_class:
+        mock_api_ctx = MagicMock()
+        plenticore_client_class.return_value = mock_api_ctx
 
+        mock_api_ctx.login = AsyncMock()
```

Comment:
If we pass `autospec=True` to the patch call above we probably don't need any of the extra mocks. We can just yield `plenticore_client_class.return_value` at the end.